### PR TITLE
[FIX] You can knock climbers off of tables again

### PR DESCRIPTION
**Pull request by Adrer**
## What Does This PR Do
You can once again knock people that are trying to climb onto a table off by clicking on the table with an empty hand.

Fixes: #27410

## Why It's Good For The Game
This feature was unintentionally removed during a fix

## Testing
Try and climb onto a table.
Click on the table.
I knock myself off.

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
fix: You can knock people climbing onto tables off by clicking on the table with an empty hand
/:cl:

### CODE REVIEW

File: code/game/objects/structures.dm [hunk 1]
```diff
@@ -99,6 +100,18 @@
 			user.visible_message("<span class='warning'>[user] leaps up onto [src]!</span>")
 		else
 			user.visible_message("<span class='warning'>[user] climbs onto [src]!</span>")
+		return TRUE
+
+/obj/structure/proc/start_climb(mob/living/user)
+	climbers += user
+	if(do_climb(user))
+		user.forceMove(get_turf(src))
+		if(get_turf(user) == get_turf(src))
```
[the remark this answers is not in the document]
Comment (Adrer) [reply code/game/objects/structures.dm:109]: It is possible for the forcemove to fail (Eg. Windoor closes in front of you)

**Review by DGamerL — APPROVED** [review]

File: code/game/objects/structures.dm [hunk 2]
```diff
@@ -93,12 +94,17 @@
 	if(!can_touch(user) || !climbable)
 		return FALSE
 
-	user.forceMove(get_turf(src))
-	if(get_turf(user) == get_turf(src))
+	return TRUE
+
+/obj/structure/proc/start_climb(mob/living/user)
+	climbers += user
```
Comment (Contrabang) [inline code/game/objects/structures.dm:100]: register a signal here to listen if the user gets deleted please, and remove them from the list if they do

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/game/objects/structures.dm [hunk 3]
```diff
@@ -98,13 +98,20 @@
...
 	climbers += user
+	RegisterSignal(user, COMSIG_PARENT_QDELETING, PROC_REF(remove_climber)) // Just in case the climber is deleted before finishing
 	if(do_climb(user))
 		user.forceMove(get_turf(src))
 		if(HAS_MIND_TRAIT(user, TRAIT_TABLE_LEAP))
 			user.visible_message("<span class='warning'>[user] leaps up onto [src]!</span>")
 		else
 			user.visible_message("<span class='warning'>[user] climbs onto [src]!</span>")
 	climbers -= user
+	UnregisterSignal(user, COMSIG_PARENT_QDELETING)
```
Comment (Contrabang) [inline code/game/objects/structures.dm:109]: maybe unregister in remove_climber too

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Review by Burzah — APPROVED** [review]

**Review by Contrabang — APPROVED** [review]
looks climbable to me

File: code/game/objects/structures.dm [hunk 4]
```diff
@@ -93,12 +94,25 @@
...
+		user.forceMove(get_turf(src))
 		if(HAS_MIND_TRAIT(user, TRAIT_TABLE_LEAP))
 			user.visible_message("<span class='warning'>[user] leaps up onto [src]!</span>")
 		else
 			user.visible_message("<span class='warning'>[user] climbs onto [src]!</span>")
+	climbers -= user
+	UnregisterSignal(user, COMSIG_PARENT_QDELETING)
+
+/obj/structure/proc/remove_climber(mob/living/climber)
+	SIGNAL_HANDLER
```
Comment (DGamerL) [inline code/game/objects/structures.dm:112]:
```suggestion
	SIGNAL_HANDLER // COMSIG_PARENT_QDELETING
```

**Review by DGamerL — APPROVED** [review]

**Pull request merged**